Automattic/jetpack
# Social: Disable `Share as a social post` if there is no media

In this change I added a simple logic that disables the share as a social post checkbox if there is no media on the post. If there is a featured image, an attached media, or SIG enabled, it won't be disabled anymore.

## Proposed changes:

* Added a new prop to the component to be able to disable it.
* Piped it back up to the form where we have all the data about media.

### Other information:


## Jetpack product discussion


https://github.com/orgs/Automattic/projects/733/views/2?pane=issue&itemId=37830733
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* Open up the editor. On a new post with SIG off, 'Share as a social post' checkbox should be disabled, and the text below should describe why is that. Do any of the following to turn off the disabling:
  * Enable SIG
  * Add a featured image
  * Add any attached media

<img width="297" alt="CleanShot 2023-09-07 at 14 26 15 png" src="https://github.com/Automattic/jetpack/assets/36671565/fb597c86-c2e3-41f1-b1fd-183ee02c002d">
